codeful-workflows/samples/recommend-coffee/codeful-logicapp-recommendcoffee/RecommendCoffee/workflow.cs

```diff
@@ -0,0 +1,184 @@
+            string composeProductListResult = string.Join(',', CoffeeArray);
+
+            var getAnEmbeddingInput = new GetSingleEmbeddingInput
+            { 
+                DeploymentId = "text-embeddings", 
+                Input = JsonSerializer.Serialize(CoffeeArray)
+            };
+
+            var embeddingResponse = await context.GetSingleEmbeddingAsync(
+                connectionId: "openai",
+                input: getAnEmbeddingInput);
+
+            //log.LogInformation("Embedding response : {result}", JsonSerializer.Serialize(embeddingResponse));
+
+            var searchVectorsInput = new VectorSearchInput 
+            { 
+                IndexName = "fourthcoffeeorders", 
+                KNearestNeighbors = 4, 
+                SearchVector = new VectorSearchInputSearchVectorType
+                { 
+                    FieldName = "embeddings", 
+                    Vector = embeddingResponse.Embedding
+                } 
+            };
+
+            var searchVectorsHttpOutput = await context.VectorSearchAsync("azureaisearch", searchVectorsInput);
+            // log.LogInformation("Vector search output : {result}", searchVectorsHttpOutputStr);
+
+            log.LogInformation("Vector search output : {result}", searchVectorsHttpOutput.First()["content"].ToString());
+
+            var chatCompletionInput = new GetChatCompletionsInput
+            {
+                DeploymentId = "gpt-4o",
+                Temperature = 1,
+                Messages = [
+                    new GetChatCompletionsInputMessagesTypeItem
+                    {
+                        Role = "System",
+                        Content = "You are an AI assistant with deep expertise in recommending coffee based on historical sales data. Focus on identifying orders where customers have purchased the same coffee as in the incoming order, and recommend additional coffee types that were frequently bought together. Prioritize recommendations from orders that include multiple products, ensuring that your suggestions are based on actual purchasing patterns. Be specific in recommending complementary coffee types that align with the customer's preferences." + 
+                        searchVectorsHttpOutput.First()["content"].ToString() + ". Please ensure that the following list of coffees are not included in the recommendations as they have already interested in them " + composeProductListResult.ToString() + ". But, please only include recommendations based upon previous Fourth Coffee purchases. There should be 3 recommendations at most. Return a json structure that returns a list of products only so that another system can read the response. call the list productreccommendations"
+                    },
+                    new GetChatCompletionsInputMessagesTypeItem
+                    {
+                        Role = "User",
+                        Content = "[\"Light Roast\"]"
```

Comment:
can we get the first message from user input rather than hardcode it in the workflow?

I would expect only the system message to be hardcoded